src/iptvsimple/data/ChannelEpg.cpp

```diff
@@ -34,8 +34,11 @@ bool ChannelEpg::UpdateFrom(const xml_node& channelNode, Channels& channels)
     return false;
 
   bool foundChannel = false;
+  bool haveDisplayNames = false;
   for (const auto& displayNameNode : channelNode.children("display-name"))
   {
+    bool haveDisplayNames = true;
```

Comment:
This doesn't make sense. I guess intention was to assign true to the var declared outside the for loop (line37)?!

Reply:
Oops, so my test would always have passed ;)

Nice find, thanks.